components/cdc/src/delegate.rs

```diff
@@ -369,9 +369,15 @@ impl Delegate {
                 CDC_PENDING_BYTES_GAUGE.add(bytes as _);
                 locks.push(PendingLock::Track { key, start_ts });
             }
-            LockTracker::Prepared { locks, .. } => match locks.entry(key) {
+            LockTracker::Prepared { locks, .. } => match locks.entry(key.clone()) {
                 BTreeMapEntry::Occupied(mut x) => {
-                    assert_eq!(x.get().ts, start_ts.ts);
+                    if x.get().ts != start_ts.ts {
+                        error!("duplicate lock with different start_ts";
```

Comment:
Please add some comments, also it feels like we do not need the fix anymore since the bug has been fixed by https://github.com/tikv/tikv/pull/18115 .

Reply:
Or we can return an error to TiCDC and terminate the rpc stream.